app/controllers/dances_controller.rb

```diff
@@ -15,9 +17,11 @@ def show
 
   def new
     @dance = Dance.new
+    @dance_direction = DanceDirection.new
   end
 
   def create
+    # NEED TO ACCOUNT FOR THE EMPTY "" STRINGS BEING ENTERED INTO THE DATABASE IN PLACE OF NIL
     @dance = Dance.new(dance_params.merge(user_id: current_user.id.to_s))
```

Comment:
In your dance model you may be able to reset your attribute to nil. For example if title is `""`

```
before_save :set_title_to_nil

def set_title_to_nil
  if self.title == ""
     self.title = nil
  end
end
```